src/circuit/transport/reservation-store.ts

```diff
@@ -162,13 +167,18 @@ export class ReservationStore extends EventEmitter<ReservationStoreEvents> imple
 
         log('created reservation on relay peer %p', peerId)
 
-        const expiration = getExpiration(reservation.expire)
+        const expiration = getExpirationMilliseconds(reservation.expire)
+
+        // sets a lower bound on the timeout
+        const timeoutDuration = Math.max(expiration - REFRESH_TIMEOUT, REFRESH_TIMEOUT_MIN)
+        // sets an upper bound on the timeout
+        const boundedTimeoutDuration = Math.min(timeoutDuration, REFRESH_TIMEOUT_MAX)
```

Comment:
If I'm reading this correctly `timeoutDuration` will either be `expiration - REFRESH_TIMEOUT` or `REFRESH_TIMEOUT_MIN`, (e.g. 30 seconds) whichever is greater.

Why do we need to then take the smaller value of `timeoutDuration` or `REFRESH_TIMEOUT_MAX ` (e.g. 15m)?

I think this means in the case where we get a reservation for an hour, we'll ignore the expiry time and refresh it after 15 minutes?

Can we not just use `timeoutDuration` having set a sensible lower bound?

Reply:
Having said that, the [go implementation](https://github.com/libp2p/go-libp2p/blob/9e208249b588727cef726a7c2560a9db722404a1/p2p/protocol/circuitv2/client/reservation.go#L89-L91) just errors out if we receive a reservation expiration in the past, maybe that's better.